From c18ead2b0c4aa62af01450cb12353a0baa51411f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bernhard=20Rosenkr=C3=A4nzer?= <bero@lindev.ch>
Date: Wed, 27 Mar 2024 12:00:18 +0100
Subject: [PATCH] Fix build with boost 1.85.0

boost::filesystem::wpath has been deprecated (and typedef-ed to
boost::filesystem::path) for a long time; it is removed from boost
starting with 1.85.0-beta1.

Use boost::filesystem::path instead.

boost/filesystem/convenience.hpp has been removed (and was being
included without being used anyway - its only use was indirectly
pulling in boost/filesystem/directory.hpp, which is actually used).

Include boost/filesystem/directory.hpp directly instead.
---
 src/core/store/MMapDirectory.cpp | 2 +-
 src/core/util/FileUtils.cpp      | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/core/store/MMapDirectory.cpp b/src/core/store/MMapDirectory.cpp
index beac7828..46156e3a 100644
--- a/src/core/store/MMapDirectory.cpp
+++ b/src/core/store/MMapDirectory.cpp
@@ -36,7 +36,7 @@ MMapIndexInput::MMapIndexInput(const String& path) {
     bufferPosition = 0;
     if (!path.empty()) {
         try {
-            file.open(boost::filesystem::wpath(path), _length);
+            file.open(boost::filesystem::path(path), _length);
         } catch (...) {
             boost::throw_exception(FileNotFoundException(path));
         }
diff --git a/src/core/util/FileUtils.cpp b/src/core/util/FileUtils.cpp
index 51508b57..d92efbb8 100644
--- a/src/core/util/FileUtils.cpp
+++ b/src/core/util/FileUtils.cpp
@@ -5,9 +5,9 @@
 /////////////////////////////////////////////////////////////////////////////
 
 #include "LuceneInc.h"
-#include <boost/filesystem/convenience.hpp>
 #include <boost/filesystem/operations.hpp>
 #include <boost/filesystem/path.hpp>
+#include <boost/filesystem/directory.hpp>
 #include "LuceneThread.h"
 #include "StringUtils.h"
 #include "FileUtils.h"
@@ -128,12 +128,12 @@ String joinPath(const String& path, const String& file) {
 }
 
 String extractPath(const String& path) {
-    boost::filesystem::wpath parentPath(path.c_str());
+    boost::filesystem::path parentPath(path.c_str());
     return parentPath.parent_path().wstring().c_str();
 }
 
 String extractFile(const String& path) {
-    boost::filesystem::wpath fileName(path.c_str());
+    boost::filesystem::path fileName(path.c_str());
     return fileName.filename().wstring().c_str();
 }
 
